chia/data_layer/data_store.py

```diff
@@ -821,3 +823,26 @@ async def get_operations(
                 previous_root = current_root
 
         return result
+
+    async def get_kv_diff(
```

Comment:
'''
``   async def get_kv_diff2(
            self, tree_id: bytes32, root_hash1: bytes32 ,root_hash2: bytes32, *, lock: bool = True
        ) -> Set[DiffData]:
            async with self.db_wrapper.locked_transaction(lock=lock):
                cursor = await self.db.execute(
                    """
                    WITH RECURSIVE
                        tree_from_root_hash(root_hash,hash, node_type, left, right, key, value) AS (
                            SELECT node.hash as root_hash, node.* FROM node WHERE node.hash == :root_hash1 or node.hash == :root_hash2
                            UNION ALL
                            SELECT
                                tree_from_root_hash.root_hash as root_hash ,node.*
                                FROM node, tree_from_root_hash
                            WHERE node.hash == tree_from_root_hash.left OR node.hash == tree_from_root_hash.right
                        )
                    SELECT * FROM tree_from_root_hash
                    WHERE node_type == :node_type
                    """,
                    {"root_hash1": root_hash1.hex(),"root_hash2": root_hash2.hex(), "node_type": NodeType.TERMINAL},
                )

                terminal_nodes_root1: List[TerminalNode] = []
                terminal_nodes_root2: List[TerminalNode] = []
                async for row in cursor:
                    root_hash = row["root_hash"]
                    node = row_to_node(row=row)
                    if not isinstance(node, TerminalNode):
                        raise Exception(f"Unexpected internal node found: {node.hash.hex()}")
                    if root_hash == root_hash1.hex():
                        terminal_nodes_root1.append(node)
                    else:
                        assert root_hash == root_hash2.hex()
                        terminal_nodes_root2.append(node)

            old_pairs = set(terminal_nodes_root1)
            new_pairs = set(terminal_nodes_root2)
            insertions = set(
                DiffData(type=OperationType.INSERT, key=node.key, value=node.value)
                for node in new_pairs
                if node not in old_pairs
            )
            deletions = set(
                DiffData(type=OperationType.DELETE, key=node.key, value=node.value)
                for node in old_pairs
                if node not in new_pairs
            )
            return set.union(insertions, deletions)`
'''

            can do this to avoid the 2 calls to get_keys_values and do only one db call
            tested with your added test test_kv_diff